### DISCUSSION TITLE
Multi env on Android native building

### DISCUSSION THREAD

**Discussion by JameDodgers** [goatandsheep/react-native-dotenv#257] [Q&A]
Hello! A few days ago I left an question [here](https://github.com/goatandsheep/react-native-dotenv/issues/209#issuecomment-976796164) and I'm posting it again just to not be forgotten. 

I have a CI that builds for different environments using Fastlane actions that under the hood use gradle and xcodebuild. As I've noticed looking over another issues, the Xcode translates the configuration names in the respective environments and so the right .env is picked up while building for a certain configuration in my Xcode project. 

On the other hand the same does not happen with **Android**. I saw that is possible to achieve this behavior changing the **build.gradle**, but I don't know how to make the **build type** choose a specific **.env** file. Could you please provide an example on how to do it?

Sorry to bother, I really need to get this and wouldn't like to resort to another library as this one seems to fully meet my other needs.

Thank you!

**Comment by goatandsheep** [comment]
Here's a bit about how build variants work: https://developer.android.com/studio/build/build-variants

Here's a bit about how signing works https://reactnative.dev/docs/signed-apk-android#adding-signing-config-to-your-apps-gradle-config

Are you struggling with use of dotenv? I can tell you a bit how that works. I noticed you are using XCode, so I'll assume you're on MacOS. To choose a specific environment, use `NODE_ENV=production npx react-native start`. See [here](https://github.com/goatandsheep/react-native-dotenv#multi-env). Is that helpful?

**Reply by JameDodgers** [reply]
Yes, I'm on Mac, but my CI runs on a Linux runner. I can't use the command you suggested because the Android build is made natively with gradlew. I'm not using build variants, but build types. One build type per environment. I have a debug, release and releaseStaging build types. So, how could I make a specific build type choose the right environment in the same way Xcode does with iOS project configurations? I've already looked at the signing example in build.gradle, but I can't see how this is helpful for me. Thank you very much for your attention!

**Reply by goatandsheep** [reply]
To be honest, I'm not sure how to do it without build variants :( we'll have to wait for someone else to help here

**Reply by JameDodgers** [reply]
I see, but what exactly do you do with build variants? Maybe I could get some hint from that. Do you have any code snippets? I don't know much about Gradle at all 🥲
